src/lib/Simulator.ts

```diff
@@ -0,0 +1,221 @@
+import { CapabilityImprovements, ContainmentPolicy } from "./PlayerActions"
+import { RandomEvent } from "./RandomEvents"
+import { NexTurnState, PlayerActions, SimulatorState, WorldSetup, WorldState } from "./SimulatorState"
+import { FakeNegativeBinomial } from "./Probabilities"
+
+export class Simulator {
+
+    private initialState: WorldSetup
+    private scaleFactor: number
+    private daysPerTurn: number
+    private currentState: WorldState
+    private history: WorldState[]
+
+    constructor(initialState: WorldSetup) {
```

Comment:
The simulator takes a scenario and uses it to create it's first version of the internal __WorldState__ using the  `computeInitialWorldState` method. A lot of this is similar to the `reset` method on `simulation.js`.